WordPress/Classes/ViewRelated/Me/MyDomains/PurchaseDomainView.swift

```diff
@@ -0,0 +1,158 @@
+import SwiftUI
+
+struct PurchaseDomainView: View {
+    private enum Constants {
+        static let imageLength: CGFloat = 36
+    }
+
+    let buyDomainAction: (() -> Void)
+    let chooseSiteAction: (() -> Void)
+
+    var body: some View {
+        VStack(alignment: .leading, spacing: Length.Padding.double) {
+            Text(Strings.header)
+                .font(.largeTitle.bold())
+            Text(Strings.subheader)
+                .font(.body)
+            getDomainCard
+            chooseSiteCard
+            Text(Strings.footnote)
+                .foregroundStyle(Color.DS.Foreground.secondary)
+                .font(.subheadline)
+            Spacer()
+        }
+        .frame(maxWidth: .infinity, alignment: .leading)
+        .padding(.horizontal, Length.Padding.double)
+        .background(Color.DS.Background.secondary)
+    }
+
+    private var getDomainCard: some View {
+        VStack(alignment: .leading, spacing: Length.Padding.single) {
+            Group {
+                Image("site-menu-domains")
+                    .renderingMode(.template)
+                    .resizable()
+                    .foregroundStyle(Color.DS.Background.brand)
+                    .frame(width: Constants.imageLength, height: Constants.imageLength)
+                    .padding(.top, Length.Padding.double)
+                Text(Strings.buyDomainTitle)
+                    .font(.title2.bold())
+                Text(Strings.buyDomainSubtitle)
+                    .foregroundStyle(Color.DS.Foreground.secondary)
+                    .padding(.bottom, Length.Padding.single)
+                PrimaryButton(title: Strings.buyDomainButtonTitle) {
+                    buyDomainAction()
+                }
+                .padding(.bottom, Length.Padding.double)
+            }
+            .padding(.horizontal, Length.Padding.double)
+        }
+        .background(Color.DS.Background.primary)
+        .clipShape(RoundedRectangle(cornerRadius: 8))
+    }
+
+    private var chooseSiteCard: some View {
+        VStack(alignment: .leading, spacing: Length.Padding.single) {
+            Group {
+                Image("block-layout")
+                    .renderingMode(.template)
+                    .resizable()
+                    .foregroundStyle(Color.DS.Background.brand)
+                    .frame(width: Constants.imageLength, height: Constants.imageLength)
+                    .padding(.top, Length.Padding.double)
+
+                chooseSiteTexts
+                    .padding(.bottom, Length.Padding.single)
+                PrimaryButton(title: Strings.chooseSiteButtonTitle) {
+                    chooseSiteAction()
+                }
+                .padding(.bottom, Length.Padding.double)
+            }
+            .padding(.horizontal, Length.Padding.double)
+        }
+        .background(Color.DS.Background.primary)
+        .clipShape(RoundedRectangle(cornerRadius: 8))
+    }
```

Comment:
[Suggestion] `chooseSiteCard` and `getDomainCard` have very similar code, so why not extracting that logic into a helper method?

```swift
private var getDomainCard: some View {
    card(imageName: "site-menu-domains",
         title: Strings.buyDomainTitle,
         subtitle: Strings.buyDomainSubtitle,
         action: buyDomainAction)
}

private var chooseSiteCard: some View {
    card(imageName: "block-layout",
         title: Strings.chooseSiteTitle,
         subtitle: Strings.chooseSiteSubtitle,
         action: chooseSiteAction) {
        Text(Strings.chooseSiteFooter)
            .foregroundStyle(Color.DS.Foreground.brand)
    }
}

private func card<Content: View>(
    imageName: String,
    title: String,
    subtitle: String,
    action: @escaping () -> Void,
    @ViewBuilder extra: () -> Content = { EmptyView() }
) -> some View {
    VStack(alignment: .leading, spacing: Length.Padding.single) {
        Group {
            Image(imageName)
                .renderingMode(.template)
                .resizable()
                .foregroundStyle(Color.DS.Background.brand)
                .frame(width: Constants.imageLength, height: Constants.imageLength)
                .padding(.top, Length.Padding.double)
            VStack(alignment: .leading, spacing: Length.Padding.single) {
                Text(title)
                    .font(.title2.bold())
                Text(subtitle)
                    .foregroundStyle(Color.DS.Foreground.secondary)
                extra()
            }
            .padding(.bottom, Length.Padding.single)
            PrimaryButton(title: Strings.buyDomainButtonTitle) {
                action()
            }
            .padding(.bottom, Length.Padding.double)
        }
        .padding(.horizontal, Length.Padding.double)
    }
    .background(Color.DS.Background.primary)
    .clipShape(RoundedRectangle(cornerRadius: 8))
}
```

Reply:
Yeah this should've been in the PR already :-] Thanks!